packages/cli/src/specs/__snapshots__/repo-builder.spec.ts.snap

```diff
@@ -78,8 +71,7 @@ export interface Todo {
 export interface TodosProps {
 }
 
-const { state, config } = createState(withProps<TodosProps>({}), withEntities<Todo>(), withUIEntities<TodoUI>(), withActiveId(), withActiveIds(), withRequestsCache<'todos'>(), withRequestsStatus<'todos'>());
-const store = new Store({ name: 'todos', state, config });
+const store = createStore({ name: 'todos' }, withProps<TodosProps>({}), withEntities<Todo>(), withUIEntities<TodoUI>(), withActiveId(), withActiveIds(), withRequestsCache<'todos'>(), withRequestsStatus<'todos'>());
```

Comment:
@EricPoul do you have something to say about this change?

Reply:
I'm not a big fan of this because it can bloat the class.

Reply:
@DmitryEfimenko, you should generally ask before doing a significant change like this. Especially when we just merged it.

Reply:
Didn't consider this to be a significant change. It's in line with the spirit of the changes in the PR. We're replacing two calls with one. The sum of code in the class will get slightly smaller. Please elaborate what you mean by saying that it will "bloat the class"

Reply:
```ts
const store = createStore({ name: 'todos' }, withProps<TodosProps>({}), withEntities<Todo>(), withUIEntities<TodoUI>(), withActiveId(), withActiveIds(), withRequestsCache<'todos'>(), withRequestsStatus<'todos'>());
```

This ^^ bloats the class code.

Reply:
I still prefer the separate method in the class and as you said it takes useful space at the top of the class. I see two ways:
1. leave it as it is now with the class method. (it generates automatically).
2. little hack with types(but it looks like a duplication of code)
```ts
private createStore(): typeof store {
  const store = createStore(....);
  
  return store;
}
```

Reply:
@DmitryEfimenko you can choose one of the options @EricPoul suggested.

Reply:
There's one more, but we had to add an a eslint ignore for method without explicit return type.

private createStore() {
  return createStore(....);
}

Reply:
Two points to counter this argument.
* I prefer the to see the store definition rather than to hide it. So having it up top would actually be preferable for me.
* The moment prettier runs, this code won't look so ugly:
```ts
store = createStore(
  { name: 'todos' },
  withProps<TodosProps>({}),
  withEntities<Todo>(),
  withUIEntities<TodoUI>(),
  withActiveId(),
  withActiveIds(),
  withRequestsCache<'todos'>(),
  withRequestsStatus<'todos'>()
);
```
Perhaps we could format the code in this way without waiting for prettier

Reply:
I do not oppose @EricPoul 's suggestion either

Reply:
I think it's annoying to see this chunk each time I need to modify the store's code. It's usually something you define once and forget about it.

Reply:
After once the prettier ran it takes a lot of space. Most of the time we don't need to know what is the store made of(if we need we jus get to the method with a help of IDE links). And it's a good practice to make a function over some quite hard logic that is responsible just for one deed. As you did with `createStore`.

And if you want to pass some data from Angular DI it will be a challenge to find it. With a separate method, we know that the store is created with some initial value from DI, and quite frequently we don't need to go into the method details.

Reply:
I'll go ahead and do this one:
```ts
private createStore(): typeof store {
  const store = createStore(....);
  return store;
}
```

Reply:
I have another reservation about this approach... We can specify the return type of the private method `createStore()`. But this does not really matter since the type of the private property `store` is not set: https://github.com/ngneat/elf/blob/master/packages/cli/src/specs/__snapshots__/repo-builder.spec.ts.snap#L26

This means that subsequent calling of `this.store.` will not give you strong typing...

To avoid this issue we could assign the store during property declaration instead of doing this in the constructor.
```
private store = this.createStore();
```

Do you have any objections?

Reply:
I went ahead and implemented the above. Let me know if you have concerns regarding this.

Reply:
It should be typed. Anyway, I'm ok with this approach. @EricPoul and you?

Reply:
I'm happy to type it. Just don't know how... the type is a complex result of calling `createStore(...)`

Reply:
It was worked in @EricPoul version.

Reply:
Did you test that it doesn't work without the typing?

https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8BQ0PQgF3gJwFMBuVdTYeAOxzwFdhc8AKAShTM02wAsBLCAHQ58BaAF4U0MAC5oARmgBfDktSKgA

Reply:
oh, you are right... [this just blew my mind](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDKAuB7ATgU2gbwLAChrQFtV4ALRAEwAoBKTAX1wZ11EhgFkBPAJVQAdEmXPj7IAlgDcw8dBCRoA3LmHRgiAHZzkAV2DyaQvPmikxEAHRyU6ALwmSZ88DTTUCazSVH8azfGiQ5NB2phZWaOZEpBSeKkwqzqiu7mgGhsaqGnLQ4bbQ6qgA7nDyqLHe0Gjw2sjqOaVe+Ex0QA). Typescript is able to figure out the type even though the property declaration and assignment are separate... hmm

Reply:
That's a new feature. It doesn't work in older versions.

Reply:
> It should be typed. Anyway, I'm ok with this approach. @EricPoul and you?

I'm ok with simple logic before the constructor. If there is a need to make something complex it's not a problem to manually move assigning in the constructor.

Reply:
updated

Reply:
I updated it